### ISSUE TITLE
fix(diffusion): masked `BiharmonicDiffusion2D/3D` (and spherical) apply the inner Laplacian unmasked — reads land values, non-conservative, coast becomes Dirichlet-0

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#292]
Part of #276.

## Summary

`BiharmonicDiffusion2D.__call__` (`finitevolx/_src/diffusion/diffusion.py:621, 648`) and the spherical version (`spherical_diffusion.py:473, 538`) compute `−κ ∇²(∇² h)` with the inner Laplacian applied without the mask. Land values therefore enter the wet tendency, the operator is not conservative on the wet domain, and the coast behaves as a Dirichlet-0 condition on ∇²h rather than no-flux. The harmonic operator is correct.

## Reproduction

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp, numpy as np
from finitevolx import CartesianGrid2D, Mask2D, BiharmonicDiffusion2D
N = 16; g = CartesianGrid2D.from_interior(N-2, N-2, Lx=1.0, Ly=1.0)
m = np.ones((N, N), bool); m[[0, -1], :] = False; m[:, [0, -1]] = False; m[6:10, 6:10] = False
mask = Mask2D.from_mask(jnp.asarray(m))
bh = BiharmonicDiffusion2D(g, mask=mask)
h = jnp.where(jnp.asarray(mask.h), 2.0, 1e10)                 # constant on the wet domain
t = bh(h, kappa=1.0)
print("max|tendency| on wet cells:", float(jnp.abs(t)[np.asarray(mask.h)].max()))   # expect 0
```

```
harmonic masked   max|t[wet]| = 0.0
biharmonic masked max|t[wet]| = 7.99999999840000e+14     <- reads land through the unmasked inner Laplacian
biharmonic masked conservation sum_wet(B q) = 67850.86   (unmasked closed operator: 3.5e-10)
change of wet tendency when land 0 -> 5: 4.0e+05
dense operator: land->wet coupling max|M[wet, land]| = 8.0e+04 ; wet-wet block max eig -264 (dissipative but wrong BC)
masked-inner -kappa·Dm(Dm(h)): symmetric, max eig -4.0e-11, land coupling 0, column sums 0, interior stencil identical
```

Golden fixtures `tests/fixtures/golden/*Biharmonic*` pin the current values.

## Root cause

`diffusion.py:621` / `:648` (2D/3D) and `spherical_diffusion.py:473` / `:538`: the inner `laplacian` call does not go through the masked no-flux stencil that the harmonic `Diffusion2D` uses.

## Proposed fix

Apply the masked harmonic operator twice: `−κ · Dm(Dm(h))` where `Dm` is the masked no-flux Laplacian (fluxes zeroed on wet→dry faces, diagonal reduced accordingly). The audit verified this variant is symmetric, negative semi-definite with the constant in its null space (max eig −4e-11), has zero land→wet coupling and zero column sums, and is bit-identical to the current operator in the interior. Regenerate goldens.

## Acceptance criteria

- [ ] Constant wet field with `land = 1e10` gives wet tendency < 1e-12 (2D, 3D, spherical).
- [ ] `Σ_wet(B q)` < 1e-10 · scale on a masked basin (conservation).
- [ ] `<p, B q> == <q, B p>` on the wet block (symmetry).
- [ ] Goldens regenerated; CHANGELOG entry.
